rules/naming/src/PhpDoc/VarTagValueNodeRenamer.php

```diff
@@ -0,0 +1,48 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Rector\Naming\PhpDoc;
+
+use PhpParser\Node;
+use PhpParser\Node\Stmt\Expression;
+use PHPStan\PhpDocParser\Ast\PhpDoc\VarTagValueNode;
+use Rector\BetterPhpDocParser\PhpDocInfo\PhpDocInfo;
+use Rector\NodeTypeResolver\Node\AttributeKey;
+
+final class VarTagValueNodeRenamer
+{
+    public function renameAssignVarTagVariableName(Node $node, string $originalName, string $expectedName): void
+    {
+        $phpDocInfo = $this->resolvePhpDocInfo($node);
+        if ($phpDocInfo === null) {
+            return;
+        }
+
+        $varTagValueNode = $phpDocInfo->getByType(VarTagValueNode::class);
+        if ($varTagValueNode === null) {
+            return;
+        }
+
+        if ($varTagValueNode->variableName !== '$' . $originalName) {
+            return;
+        }
+
+        $varTagValueNode->variableName = '$' . $expectedName;
+    }
+
+    /**
+     * Expression doc block has higher priority
+     */
+    private function resolvePhpDocInfo(Node $node): ?PhpDocInfo
+    {
+        $phpDocInfo = $node->getAttribute(AttributeKey::PHP_DOC_INFO);
+
+        $parent = $node->getAttribute(AttributeKey::PARENT_NODE);
+        if ($parent instanceof Expression) {
+            $expressionPhpDocInfo = $parent->getAttribute(AttributeKey::PHP_DOC_INFO);
+        }
```

Comment:
@dobryy So this was the issue :face_with_head_bandage: 

The PhpDocInfo is used twice, so it's accessible on the `Assign` node (so you don't have to always traverse up to `Expression`).

<br>

`Assign` (no `;` in the end)

```php
/** @var Node $name */
$name = $this->getName();
```

That was printer correctly. But overriden in the print by parent expression.

<br>

`Expression`

```php
/** @var Node $a */
$name = $this->getName();
```

Only the `Expression` is printed.

Reply:
After I decopuled all classes I could (~3 hours), it took me around 20 minutes to find this one. It's easier for me to work only with current minimal scope of what is going on.